hash_practice/exercises.py

```diff
@@ -1,19 +1,42 @@
-
+from collections import defaultdict, deque
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(m (n log n)) where m is number of words & n is avg number of characters in chars in each word

```

Comment:
I think your answer is a more accurate answer than we're accepting, but we are also accepting O(m). Making the assumption these are English words, we can assume they tend not to get too long in length (on average 4.7 letters per word from a quick search). Especially as the number of words goes up to the hundreds or possibly thousands, that number is going to be dwarfed.

Nothing wrong, just letting you know.

Reply:
Oh and thank you so much for the detailed explanations of how you did your calculations! It's very helpful for me not having to guess what you're thinking. :3